tests/Plugin/InvokeStrategy/OnEventStrategyTest.php

```diff
@@ -34,4 +37,23 @@ public function it_invokes_the_on_event_method_of_the_handler(): void
 
         $this->assertSame($customEvent, $onEventHandler->getLastMessage());
     }
+
+    /**
+     * @test
+     */
+    public function it_can_be_attached_to_event_bus(): void
+    {
+        $onEventStrategy = new OnEventStrategy();
+
+        $prophet = new \Prophecy\Prophet();
+
+        $bus = $prophet->prophesize(EventBus::class);
+        $bus->attach(Argument::type('string'), Argument::type('callable'), Argument::type('integer'))->willReturn(new DefaultListenerHandler(function () {
+        }));
+
+        $onEventStrategy->attachToMessageBus($bus->reveal());
+
+        $bus->attach(Argument::type('string'), Argument::type('callable'), Argument::type('integer'))->shouldHaveBeenCalled();
+        $prophet->checkPredictions();
```

Comment:
Instead of manuell run checkPredictions(), you could run
```php
$bus->attach(Argument::type('string'), Argument::type('callable'), Argument::type('integer'))
    ->shouldBeCalled()
    ->willReturn(new DefaultListenerHandler(function () {}));
```

Then checkPredictions() would be executed from PHPUnit.